### PR TITLE
config-tools: resolve security issues

### PR DESCRIPTION
**Pull request by yuchuyang**

### CODE REVIEW

**Comment by yuchuyang** [comment]
@wenlingz This PR need to install a new package using "pip3 install defusedxml". It replaces the xml package regarding the security issue. To find the details of defusedxml: https://pypi.org/project/defusedxml/.

**Comment by dbkinder** [comment]
Requiring a new python package is also a documentation change @NanlinXie

**Review by dbkinder — CHANGES_REQUESTED** [review]
updating the docs LGTM.  You'll also need to update ./misc/config_tools/config_app/requirements, right?

File: misc/config_tools/board_inspector/cli.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 import sys, os
 import logging
-import subprocess
+import subprocess #nosec
```
Comment (dbkinder) [inline misc/config_tools/board_inspector/cli.py:10]: Can you verify that a space isn't required, ``#nosec`` vs. ``# nosec`` (the bandit documentation says to use a space).

File: doc/getting-started/building-from-source.rst [hunk 2]
```diff
@@ -80,7 +80,7 @@ ACRN.
           xsltproc \
           clang-format
 
-     sudo pip3 install lxml xmlschema
+     sudo pip3 install lxml xmlschema defusedxml
```
Comment (gvancuts) [inline doc/getting-started/building-from-source.rst:83]: Do we still need `lxml`? If not, please remove it from the list.

File: doc/getting-started/getting-started.rst [hunk 3]
```diff
@@ -180,7 +180,7 @@ Build the ACRN Hypervisor on Ubuntu
         xsltproc \
         clang-format
 
-      $ sudo pip3 install lxml xmlschema
+      $ sudo pip3 install lxml xmlschema defusedxml
```
Comment (gvancuts) [inline doc/getting-started/getting-started.rst:183]: Same question about `lxml`

File: misc/config_tools/config_app/requirements [hunk 4]
```diff
@@ -2,3 +2,4 @@ Flask==1.1.1
 flask_bootstrap==3.3.7.1
 xmlschema==1.4.1
 lxml==4.6.2
+defusedxml==0.7.1
```
Comment (gvancuts) [inline misc/config_tools/config_app/requirements:5]: Ditto, can we remove `lxml` from the file?

**Review by gvancuts — COMMENTED** [review]
It looks like `lxml` remains part of the `board_inspector` code, is that expected?

**Comment by yuchuyang** [comment]
@gvancuts lxml is still widely used in config-tools.

Bandit does recommend to replace the lxml using defusedxml.lxml; however, lxml is deprecated based and may be removed in the future, please check the below links:
https://github.com/tiran/defusedxml/issues/38
https://github.com/tiran/defusedxml/blob/main/defusedxml/lxml.py

After talked with security team, I thought we may give it a try to work on improving the code with the idea of defusedxml.lxml but without directly using the module from defusedxml.

TLDR, I am working on the lxml security issue and the lxml is still in use.

File: misc/config_tools/config_app/requirements (hunk 4 above)
Comment (dbkinder) [reply misc/config_tools/config_app/requirements:5]: What this shows too is we have this list of python dependencies in too many places.  Best is to put them in one ``requirements.txt`` file and use that everywhere.

**Review by gvancuts — APPROVED** [review]

File: misc/config_tools/config_app/requirements (hunk 4 above)
Comment (dbkinder) [reply misc/config_tools/config_app/requirements:5]: @yuchuyang We can do this "single list in one place" fix in another PR if you need to merge this one for continuing your security work.

**Review by NanlinXie — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]

**Comment by wenlingz** [comment]
@yuchuyang please rebase the code, thanks

**Comment by yuchuyang** [comment]
Still failed with " ModuleNotFoundError: No module named 'defusedxml'"

**Review by yonghuah — COMMENTED** [review]
fine to me, thanks.
